src/wasm-validator.h

```diff
@@ -46,13 +46,15 @@
 namespace wasm {
 
 // Print anything that can be streamed to an ostream
-template <typename T>
+  template <typename T,
+    typename std::enable_if<
+      !std::is_base_of<Expression, typename std::remove_pointer<T>::type>::value
```

Comment:
We could do the wrapping in the calls to `fail` but yeah, maybe that's no better.